### ISSUE TITLE
if the mail text contains a single quote, action log insertion fails

### ISSUE THREAD

**Issue by ahelsing** [GENI-NSF/geni-ar#42]
Steps to reproduce:
- Request an account
- From the admin UI, select "EMAIL REQUESTER" and click "SUBMIT"
- Modify the message to contain one or more single quotes, e.g.:

```
Dear Chaos,

Your request for a GENI IdP account cannot be fulfilled because we like 'single quotes'.

Please...

Thanks,
The GENI Project Office

```
- Click "SEND"

You will see:

```
Couldn't create log. Postgres database insert failed
```

at [https://shib-idp1.gpolab.bbn.com/manage/send_email.php], the apache log will contain:

```
[Tue Sep 17 15:50:19 2013] [error] [client 128.89.73.13] Couldn't create log. Postgres database insert failed, referer: https://shib-idp1.gpolab.bbn.com/manage/request_actions.php
```

and there will be no log entry corresponding to the sent request, so the UI will not be able to display which admin sent the e-mail.

 _Imported from trac ticket [#42](http://trac.gpolab.bbn.com/idp-tools/ticket/42),  created by chaos on 09-17-2013 at 15:51, last modified: 10-03-2013 at 16:51_

**Comment by ahelsing** [comment]
Whew.  Just in case you were wondering why i was doing so many crazy things on the system leading to the previous two tickets, it's because i was trying to reproduce this bug.  It took me a number of tries to think of the single quote issue.

IMO, this one is a blocker --- people might put all kinds of weird characters in e-mails to requesters, and SQL will choke on some subset of them if left to its own devices.

_Trac comment by chaos on 09-17-2013 at 15:53_

**Comment by ahelsing** [comment]
I googled and the postgres has problems with single quotes and backslashes.  They need to be doubled.  So I added str_replace to double them before creating the log.  To be extra safe, in case some other weird character causes a problem, if the log fails I also added a second try to log without including the email text in the comment.  That way at least there is a record of which admin sent the email and when. 

_Trac comment by phelinek on 09-18-2013 at 14:27_
